examples/alerts/rules.yaml

```diff
@@ -0,0 +1,151 @@
+groups:
+- name: thanos-querier.rules
+  rules:
+  - expr: |
+      (
+        sum(rate(grpc_client_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable|DataLoss|DeadlineExceeded", job=~"thanos-querier.*", grpc_type="unary"}[5m]))
+      /
+        sum(rate(grpc_client_started_total{job=~"thanos-querier.*", grpc_type="unary"}[5m]))
+      )
+    labels: {}
+    record: :grpc_client_failures_per_unary:sum_rate
+  - expr: |
+      (
+        sum(rate(grpc_client_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable|DataLoss|DeadlineExceeded", job=~"thanos-querier.*", grpc_type="server_stream"}[5m]))
+      /
+        sum(rate(grpc_client_started_total{job=~"thanos-querier.*", grpc_type="server_stream"}[5m]))
+      )
+    labels: {}
+    record: :grpc_client_failures_per_stream:sum_rate
+  - expr: |
+      (
+        sum(rate(thanos_querier_store_apis_dns_failures_total{job=~"thanos-querier.*"}[5m]))
+      /
+        sum(rate(thanos_querier_store_apis_dns_lookups_total{job=~"thanos-querier.*"}[5m]))
+      )
+    labels: {}
+    record: :thanos_querier_store_apis_dns_failures_per_lookup:sum_rate
+  - expr: |
+      histogram_quantile(0.99,
+        sum(http_request_duration_seconds_bucket{job=~"thanos-querier.*", handler="query"}) by (le)
+      )
+    labels:
+      quantile: "0.99"
+    record: :query_duration_seconds:p99:histogram_quantile
+  - expr: |
+      histogram_quantile(0.99,
+        sum(rate(http_request_duration_seconds_bucket{job=~"thanos-querier.*", handler="query"}[5m])) by (le)
+      )
+    labels:
+      quantile: "0.99"
+    record: :query_duration_seconds:histogram_quantile
+  - expr: |
+      histogram_quantile(0.99,
+        sum(http_request_duration_seconds_bucket{job=~"thanos-querier.*", handler="query_range"}) by (le)
+      )
+    labels:
+      quantile: "0.99"
+    record: :api_range_query_duration_seconds:histogram_quantile
+  - expr: |
+      histogram_quantile(0.99,
+        sum(rate(http_request_duration_seconds_bucket{job=~"thanos-querier.*", handler="query_range"}[5m])) by (le)
+      )
+    labels:
+      quantile: "0.99"
+    record: :api_range_query_duration_seconds:histogram_quantile
+- name: thanos-receive.rules
+  rules:
+  - expr: |
+      sum(
+        rate(grpc_server_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable|DataLoss|DeadlineExceeded", job=~"thanos-querier.*", grpc_type="unary"}[5m])
+      /
+        rate(grpc_server_started_total{job=~"thanos-querier.*", grpc_type="unary"}[5m])
+      )
+    labels: {}
+    record: :grpc_server_failures_per_unary:sum_rate
```

Comment:
Isn't something missing here before `:` and in other places?

Reply:
+1 this looks like a bug

Reply:
This is another convention if the metric doesn't provide any aggregated label. e.g https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/13efb4da7a53add764bbbba9d17efb895ba5bcac/rules/node.libsonnet#L40-L41

However, there are other conventions, I can have another look at it.

Reply:
Yes, this is the convention if `sum(rate(` doesn't aggregate by anything.